app/controllers/concerns/scim_rails/response.rb

```diff
@@ -24,6 +24,23 @@ def json_scim_response(object:, status: :ok, counts: nil)
       end
     end
 
+    # TODO: Cut down on redundant code in this file
+
+    def json_scim_group_response(object:, status: :ok, counts: nil)
+      case params[:action]
+      when "index"
+        render \
+          json: list_group_response(object, counts),
+          status: status,
+          content_type: CONTENT_TYPE
+      when "show", "create", "put_update", "patch_update"
+        render \
+          json: group_response(object),
+          status: status,
+          content_type: CONTENT_TYPE
+      end
```

Comment:
seems like the only difference is the thing getting rendered here.
cleaner to use the case/switch to set a var for the result and have one render block